<a href="https://colab.research.google.com/github/todebode/Garments-employee-productivity/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 

Tijesunimi Odebode

03/13/2022

## **Part 1 - Loading and verifying data**

Each of the following tasks will be addressed:

1. Import necessary packages

2. Load and verify data

In [98]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


### 1. Import necessary packages

In [99]:
import pandas as pd

### 2. Load and verify data

### **Data 1**

In [100]:
filename = '/content/garments_worker_productivity (1).csv'
df = pd.read_csv(filename)
df.head()

date   quarter  department       day  team  targeted_productivity  \
0  1/1/2015  Quarter1      sweing  Thursday     8                   0.80   
1  1/1/2015  Quarter1  finishing   Thursday     1                   0.75   
2  1/1/2015  Quarter1      sweing  Thursday    11                   0.80   
3  1/1/2015  Quarter1      sweing  Thursday    12                   0.80   
4  1/1/2015  Quarter1      sweing  Thursday     6                   0.80   

     smv     wip  over_time  incentive  idle_time  idle_men  \
0  26.16  1108.0       7080         98        0.0         0   
1   3.94     NaN        960          0        0.0         0   
2  11.41   968.0       3660         50        0.0         0   
3  11.41   968.0       3660         50        0.0         0   
4  25.90  1170.0       1920         50        0.0         0   

   no_of_style_change  no_of_workers  actual_productivity  
0                   0           59.0             0.940725  
1                   0            8.0             0.886500  
2                   0           30.5             0.800570  
3                   0           30.5             0.800570  
4                   0           56.0             0.800382

In [101]:
# Let's determine how many features in our data set
X = df.drop(columns=['actual_productivity'])
y = df['actual_productivity']
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1197 entries, 0 to 1196
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   date                   1197 non-null   object 
 1   quarter                1197 non-null   object 
 2   department             1197 non-null   object 
 3   day                    1197 non-null   object 
 4   team                   1197 non-null   int64  
 5   targeted_productivity  1197 non-null   float64
 6   smv                    1197 non-null   float64
 7   wip                    691 non-null    float64
 8   over_time              1197 non-null   int64  
 9   incentive              1197 non-null   int64  
 10  idle_time              1197 non-null   float64
 11  idle_men               1197 non-null   int64  
 12  no_of_style_change     1197 non-null   int64  
 13  no_of_workers          1197 non-null   float64
dtypes: float64(5), int64(5), object(4)
memory usage: 131.0+ 

- There are 14 columns representing 14 features.

In [102]:
# Let's determine how many rows and missing data in our data set
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1197 entries, 0 to 1196
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   date                   1197 non-null   object 
 1   quarter                1197 non-null   object 
 2   department             1197 non-null   object 
 3   day                    1197 non-null   object 
 4   team                   1197 non-null   int64  
 5   targeted_productivity  1197 non-null   float64
 6   smv                    1197 non-null   float64
 7   wip                    691 non-null    float64
 8   over_time              1197 non-null   int64  
 9   incentive              1197 non-null   int64  
 10  idle_time              1197 non-null   float64
 11  idle_men               1197 non-null   int64  
 12  no_of_style_change     1197 non-null   int64  
 13  no_of_workers          1197 non-null   float64
 14  actual_productivity    1197 non-null   float64
dtypes: f

- There are 1197 rows and there is missing data in the **'wip'** column.

**1) Source of data**

*UCI Machine Learning Repository: https://archive.ics.uci.edu/ml/datasets/Productivity+Prediction+of+Garment+Employees*

**2) Brief description of data**

The dataset contains some important attributes of the garment manufacturing process and employee productivity performance. (https://archive.ics.uci.edu/ml/datasets/Productivity+Prediction+of+Garment+Employees)

**3) What is the target?**

The target variable is 'actual_productivity.'

**4) Is this a classification or regression problem?** 

I will divide the productivity ranges ('actual_productivity' variable) into different classes and make this a classification problem where I predict different productivity ranges.

**5) How many features?**

There are 14 features

**6) How many rows of data.**

There are 1197 rows based on 'df.info()' above

**7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?**
I really don't foresee any major challenges. I would have to deal with missing values though. I would also have to divide productivity ranges into different classes.

### **Data 2**

In [103]:
filename = '/content/Wholesale customers data.csv'
df = pd.read_csv(filename)
df.head()


Channel  Region  Fresh  Milk  Grocery  Frozen  Detergents_Paper  Delicassen
0        2       3  12669  9656     7561     214              2674        1338
1        2       3   7057  9810     9568    1762              3293        1776
2        2       3   6353  8808     7684    2405              3516        7844
3        1       3  13265  1196     4221    6404               507        1788
4        2       3  22615  5410     7198    3915              1777        5185

In [104]:
# Let's determine how many features in our data set
X = df.drop(columns=['Channel'])
y = df['Channel']
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 440 entries, 0 to 439
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Region            440 non-null    int64
 1   Fresh             440 non-null    int64
 2   Milk              440 non-null    int64
 3   Grocery           440 non-null    int64
 4   Frozen            440 non-null    int64
 5   Detergents_Paper  440 non-null    int64
 6   Delicassen        440 non-null    int64
dtypes: int64(7)
memory usage: 24.2 KB


- There are 7 columns representing 7 features.

In [105]:
# Let's determine how many rows and missing data in our data set
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 440 entries, 0 to 439
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Channel           440 non-null    int64
 1   Region            440 non-null    int64
 2   Fresh             440 non-null    int64
 3   Milk              440 non-null    int64
 4   Grocery           440 non-null    int64
 5   Frozen            440 non-null    int64
 6   Detergents_Paper  440 non-null    int64
 7   Delicassen        440 non-null    int64
dtypes: int64(8)
memory usage: 27.6 KB


- There are 440 rows and there is no missing data in this dataset.

**1) Source of data**

*UCI Machine Learning Repository: https://archive.ics.uci.edu/ml/datasets/Wholesale+customers

**2) Brief description of data**

The dataset contains clients of a wholesale distributor and the annual spending in monetary units (m.u.) on diverse product categories (https://archive.ics.uci.edu/ml/datasets/Wholesale+customers)

**3) What is the target?**

The target variable is 'Channel.'

**4) Is this a classification or regression problem?** 

This will be a classification problem.

**5) How many features?**

There are 7 features

**6) How many rows of data.**

There are 440 rows based on 'df.info()' above

**7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?**
I really don't foresee any major challenges. 

Notes for Data 2: The two types of 'Channel' are Horeca (Hotel/Restaurant/Café) and Retail. Each type is represented by 1 or 2. The one with the higher frequency (298) is '1.' The one with the lower frequency (142) is '2.' Confirm this using a code to see how many 1s or 2s are in that column. The frequency information is already given n the UCI website as well. So Horeca is represented by 1 and Retail is represented by 2.